src/vs/workbench/browser/workbench.contribution.ts

```diff
@@ -85,6 +86,33 @@ const registry = Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Con
 				'markdownDescription': localize('decorations.colors', "Controls whether editor file decorations should use colors."),
 				'default': true
 			},
+			[CustomEditorLabelService.SETTING_ID_ENABLED]: {
+				'type': 'boolean',
+				'markdownDescription': localize('workbench.editor.label.enabled', "Controls whether the custom workbench editor labels should be applied."),
+				'default': true,
+			},
+			[CustomEditorLabelService.SETTING_ID_PATTERNS]: {
+				'type': 'object',
+				scope: ConfigurationScope.RESOURCE,
+				'markdownDescription': (() => {
+					let customEditorLabelDescription = localize('workbench.editor.label.patterns', "Controls the rendering of the editor label. Each __Item__ is a pattern that matches a file path. Both relative and absolute file paths are supported. In case multiple patterns match, the longest matching path will be picked. Each __Value__ is the template for the rendered editor when the __Item__ matches. Variables are substituted based on the context:");
+					customEditorLabelDescription += '\n- ' + [
+						localize('workbench.editor.label.dirname', "`${dirname}`: name of the folder in which the file is located (e.g. root/folder/file.txt -> folder)."),
```

Comment:
Is it possible to put backticks around paths here as well to make it easier to read the setting?